### PR TITLE
Add CircleCI and go mod turds

### PR DESCRIPTION
**Pull request by anacrolix**
Add CircleCI testing of go mod compatibility. I had random test failures while mucking about and produced this in trying to resolve the issue.

### CODE REVIEW

**Comment by hsanjuan** [comment]
It's more like "Adding go.mod" (plus circle) ?

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	github.com/libp2p/go-libp2p-host v3.0.15+incompatible
+	github.com/libp2p/go-libp2p-interface-connmgr v0.0.21 // indirect
+	github.com/libp2p/go-libp2p-interface-pnet v3.0.0+incompatible // indirect
+	github.com/libp2p/go-libp2p-loggables v1.1.24 // indirect
+	github.com/libp2p/go-libp2p-metrics v2.1.7+incompatible // indirect
+	github.com/libp2p/go-libp2p-nat v0.8.8 // indirect
+	github.com/libp2p/go-libp2p-net v3.0.15+incompatible
+	github.com/libp2p/go-libp2p-peer v2.4.0+incompatible
+	github.com/libp2p/go-libp2p-peerstore v2.0.6+incompatible
+	github.com/libp2p/go-libp2p-protocol v1.0.0
```
Comment (hsanjuan) [inline go.mod]: why is this not indirect. shouldn't we just have a direct import being top-level libp2p?

**Comment by raulk** [comment]
I defer to @hsanjuan here.

File: go.mod (hunk 1 above)
Comment (anacrolix) [reply go.mod]: I don't follow. the protocol import?

**Comment by anacrolix** [comment]
@hsanjuan let's get this merged.

**Comment by hsanjuan** [comment]
Is there anything depending on this other than cluster? We don't need it right now, and when we do we will have to set the right deps anyway, so I'd rather defer it until I have gained some further knowledge about go modules and libp2p has officially adopted go modules (and done the tag rewriting).

**Comment by anacrolix** [comment]
There are no tags in this repo. go mod is designed to allow incrementally including support. This is one such example, including go mod here will allow downstream repos to incrementally test their ability to use a dependency with go mod support like this one. If the go mod needs to change, it's part of the design that downstream won't break when we do that.

**Comment by hsanjuan** [comment]
Right, so since downstream is just cluster, I'll do this when needed, once libp2p officially supports modules and tags are set and verifying myself the resulting dependency tree.

By the way, can you disable circleCI? There was no need to add circle where travis works.

**Comment by anacrolix** [comment]
> By the way, can you disable circleCI? There was no need to add circle where travis works.

Travis doesn't verify go modules.

**Comment by hsanjuan** [comment]
I'm not sure what you mean by "verify" because I cannot see anything special happening in circle that is outside the regular go pipeline and which travis could do.  And this has created a broken status check for all branches that was never asked for..

**Comment by hsanjuan** [comment]
Fixed by #43

**Pull request closed without merging**